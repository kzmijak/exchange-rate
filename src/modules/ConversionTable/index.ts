export { ConversionTable } from "./src/ConversionTable";
