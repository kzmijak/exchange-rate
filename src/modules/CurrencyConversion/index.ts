export { useCurrencyConverter } from "./src/hooks/useCurrencyConverter";
