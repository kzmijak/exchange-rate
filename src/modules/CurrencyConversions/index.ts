export { ConversionRate } from "./src/mobx/ConversionRate";
