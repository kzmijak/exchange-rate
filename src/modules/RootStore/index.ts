export { RootStoreProvider } from "./src/contexts/RootStoreContext";
export { useRootStore } from "./src/hooks/useRootStore";
export { createRootStore } from "./src/rootStore";
export type { RootStore } from "./src/rootStore";
