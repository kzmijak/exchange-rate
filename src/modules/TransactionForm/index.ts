export { TransactionForm } from "./src/components/TransactionForm";
export type { TransactionFormContent } from "./src/models/TransactionFormContent";
