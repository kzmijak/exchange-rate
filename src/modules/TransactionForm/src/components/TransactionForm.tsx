import { FC } from "react";
import { TransactionFormContent } from "../models/TransactionFormContent";
import { useForm } from "react-hook-form";
import { yupResolver } from "@hookform/resolvers/yup";
import { schema } from "../utils/schema";
import { Button, Input, Stack, Typography } from "@mui/joy";

type TransactionFormProps = {
  defaultValues?: Partial<TransactionFormContent>;
  onSubmit: (formContent: TransactionFormContent) => void;
};

export const TransactionForm: FC<TransactionFormProps> = ({
  defaultValues,
  onSubmit,
}) => {
  const { register, handleSubmit } = useForm<TransactionFormContent>({
    defaultValues,
    resolver: yupResolver(schema),
  });

  return (
    <Stack component="form" onSubmit={handleSubmit(onSubmit)}>
      <Stack flex={1}>
        <Stack direction="row">
          <Typography>Title of transaction</Typography>
          <Input {...register("title")} />
        </Stack>
        <Stack direction="row">
          <Typography>Amount (in PLN)</Typography>
          <Input {...register("amount")} />
        </Stack>
      </Stack>
      <Button type="submit">Add</Button>
    </Stack>
  );
};
