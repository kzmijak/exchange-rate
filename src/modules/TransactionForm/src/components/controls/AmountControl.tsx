import { FC } from "react";
import { Control, useController } from "react-hook-form";
import { TransactionFormContent } from "../../..";
import { Label } from "../utils/Label";
import { FormHelperText, Input } from "@mui/joy";

type AmountControlProps = {
  control: Control<TransactionFormContent>;
};

export const AmountControl: FC<AmountControlProps> = ({ control }) => {
  const { field, fieldState } = useController({
    control,
    name: "amount",
  });

  return (
    <Label label="Amount (in PLN)">
      <Input
        {...field}
        fullWidth
        type="number"
        placeholder="ex. 25,99"
        endDecorator="zÅ‚"
      />
      {fieldState.error && (
        <FormHelperText>{fieldState.error.message}</FormHelperText>
      )}
    </Label>
  );
};
