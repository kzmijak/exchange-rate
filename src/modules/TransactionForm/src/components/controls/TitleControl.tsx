import { FC } from "react";
import { Control, useController } from "react-hook-form";
import { TransactionFormContent } from "../../..";
import { Label } from "../utils/Label";
import { FormHelperText, Input } from "@mui/joy";

type TitleControlProps = {
  control: Control<TransactionFormContent>;
};

export const TitleControl: FC<TitleControlProps> = ({ control }) => {
  const { field, fieldState } = useController({
    control,
    name: "title",
  });

  return (
    <Label label="Title of transaction">
      <Input {...field} fullWidth placeholder="ex. Busk ticket" />
      {fieldState.error && (
        <FormHelperText color="danger.500">
          {fieldState.error.message}
        </FormHelperText>
      )}
    </Label>
  );
};
