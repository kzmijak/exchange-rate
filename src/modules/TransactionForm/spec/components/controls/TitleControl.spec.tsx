import { describe, it, expect } from "vitest";
import { render, renderHook } from "@testing-library/react";
import { TitleControl } from "../../../src/components/controls/TitleControl";
import { useForm } from "react-hook-form";
import { TransactionFormContent } from "modules/TransactionForm";

describe("TitleControl", () => {
  const { result } = renderHook(() => useForm<TransactionFormContent>());
  const Component = () => <TitleControl control={result.current.control} />;

  it("should match snapshot", () => {
    expect(render(<Component />).container).toMatchSnapshot();
  });
});
