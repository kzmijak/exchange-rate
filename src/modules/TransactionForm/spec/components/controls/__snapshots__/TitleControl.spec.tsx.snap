// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TitleControl > should match snapshot 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiFormControl-vertical MuiFormControl-sizeMd MuiStack-root css-9cy8of-JoyFormControl-root-JoyStack-root"
  >
    <label
      class="MuiFormLabel-root css-1ftzm8l-JoyFormLabel-root"
      for=":r0:"
      id=":r0:-label"
    >
      Title of transaction
    </label>
    <div
      class="MuiBox-root css-1rr4qq7"
    >
      <div
        class="MuiInput-root MuiInput-fullWidth MuiInput-variantOutlined MuiInput-colorNeutral MuiInput-sizeMd Joy-focused MuiInput-formControl css-gsnhoa-JoyInput-root"
      >
        <input
          class="MuiInput-input css-1gw9vc6-JoyInput-input"
          id=":r0:"
          name="title"
          placeholder="ex. Bus ticket"
          value=""
        />
      </div>
    </div>
  </div>
</div>
`;
